### PR TITLE
add customStateCheck into api doc

### PR DESCRIPTION
**Pull request by ruanxin**
related issue: https://github.com/kyma-project/lifecycle-manager/issues/196

### CODE REVIEW

**Review by medmes — DISMISSED** [review]
Looks good for me, I also double checked the spelling and grammar of the texts!

File: docs/technical-reference/api/moduleTemplate-cr.md [hunk 1]
```diff
@@ -45,6 +45,42 @@ spec:
 
 If not specified, the **namespace** of the resource mentioned in **.spec.data** will be controlled by the `sync-namespace` flag; otherwise, it will be respected. All other attributes (including **.metadata.name**, **apiVersion**, and **kind**) are taken over as stated. Note that since it behaves similarly to a `template`, any subresources, such as **status**, are ignored, even if specified in the field.
 
+### **.spec.customStateCheck**
+
+The `.spec.customStateCheck` field in Kyma Lifecycle Manager is primarily designed for third-party modules. For non-Kyma modules, the `status.state` might not be present, which the lifecycle manager relies on to determine module state. This field enables users to define custom fields in the module Custom Resource (CR) that can be mapped to valid states supported by the lifecycle manager.
```
Comment (grego952) [inline docs/technical-reference/api/moduleTemplate-cr.md:50]:
```suggestion
The `.spec.customStateCheck` field in Kyma Lifecycle Manager is primarily designed for third-party modules. For non-Kyma modules, the `status.state` might not be present, which the Lifecycle Manager relies on to determine the module state. This field enables users to define custom fields in the module Custom Resource (CR) that can be mapped to valid states supported by Lifecycle Manager.
```

File: docs/technical-reference/api/moduleTemplate-cr.md [hunk 2]
```diff
@@ -45,6 +45,42 @@ spec:
...
+
+Imagine a scenario where a module's health is indicated by status.health in its CR. In such cases, users can employ the customStateCheck configuration to map the health states to lifecycle manager states.
+
+Here's an example YAML configuration:
```
Comment (grego952) [inline docs/technical-reference/api/moduleTemplate-cr.md:54]:
```suggestion
Here's an example of YAML configuration:
```

File: docs/technical-reference/api/moduleTemplate-cr.md [hunk 3]
```diff
@@ -45,6 +45,42 @@ spec:
...
+  customStateCheck:
+  - jsonPath: 'status.health'
+    value: 'green'
+    mappedState: 'Ready'
+  - jsonPath: 'status.health'
+    value: 'red'
+    mappedState: 'Error'
+```
+
+In this example, when the module's CR is in the green health state, the corresponding Kyma CR will transition to the Ready state. Similarly, when the module's CR is in the red health state, the related Kyma CR will transition to the Error state.
```
Comment (grego952) [inline docs/technical-reference/api/moduleTemplate-cr.md:66]:
```suggestion
In this example, when the module's CR is in the green health state, the corresponding Kyma CR will transition to the `Ready` state. Similarly, when the module's CR is in the red health state, the related Kyma CR will transition to the `Error` state.
```

**Review by medmes — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Review by medmes — APPROVED** [review]

**Pull request merged**